src/v/iceberg/tests/manifest_serialization_test.cc

```diff
@@ -0,0 +1,253 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "base/vlog.h"
+#include "bytes/iobuf.h"
+#include "iceberg/manifest_entry.h"
+#include "iceberg/manifest_file.h"
+
+#include <seastar/core/temporary_buffer.hh>
+
+#include <avro/Stream.hh>
+#include <gtest/gtest.h>
+
+using namespace iceberg;
+
+namespace {
+
+ss::logger icelog("icebergtest");
+
+// Near-identical implementation of avro::MemoryOutputStream, but backed by an
+// iobuf that can be released.
+class avro_iobuf_ostream : public avro::OutputStream {
+public:
+    explicit avro_iobuf_ostream(size_t chunk_size)
+      : chunk_size_(chunk_size)
+      , available_(0)
+      , byte_count_(0) {}
+    ~avro_iobuf_ostream() override = default;
+
+    iobuf&& release() {
+        buf_.trim_back(buf_.size_bytes() - byte_count_);
+        available_ = 0;
+        byte_count_ = 0;
+        vlog(icelog.trace, "ostream::release({})", buf_.size_bytes());
+        return std::move(buf_);
+    }
```

Comment:
this should be `iobuf&& release() && {` r-value qualified. or instead, if you want to reuse the ostream, dont' return r-value reference plus `return std::exchange(buf, iobuf());` or something equivalent?

Reply:
Yeah good point. Done, went for the latter.